sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -698,41 +670,43 @@ export class TableClient {
    * await client.updateEntity(entity, "Replace")
    * ```
    */
-  public async updateEntity<T extends object>(
+  public updateEntity<T extends object>(
     entity: TableEntity<T>,
     mode: UpdateMode = "Merge",
     // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
     options: UpdateTableEntityOptions = {}
   ): Promise<UpdateEntityResponse> {
-    const { span, updatedOptions } = createSpan(`TableClient-updateEntity-${mode}`, options);
-
-    try {
-      const partitionKey = escapeQuotes(entity.partitionKey);
-      const rowKey = escapeQuotes(entity.rowKey);
-
-      const { etag = "*", ...updateEntityOptions } = updatedOptions || {};
-      if (mode === "Merge") {
-        return await this.table.mergeEntity(this.tableName, partitionKey, rowKey, {
-          tableEntityProperties: serialize(entity),
-          ifMatch: etag,
-          ...updateEntityOptions,
-        });
-      }
-      if (mode === "Replace") {
-        return await this.table.updateEntity(this.tableName, partitionKey, rowKey, {
-          tableEntityProperties: serialize(entity),
-          ifMatch: etag,
-          ...updateEntityOptions,
-        });
-      }
+    return tracingClient.withSpan(
+      "TableClient.updateEntity",
```

Comment:
Calling out a change here - instead of `TableClient-updateEntity-${mode}` I am using `TableClient.updateEntity` and adding the updateEntityMode as a span attribute below.

This follows the guidelines to use low-cardinality span names where possible and aligns with our naming conventions. Same for upsertEntity